### ISSUE TITLE
Type error on strict compiling

### ISSUE THREAD

**Issue by chrisch-hh** [larscom/ngrx-store-storagesync#20]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

Not quite sure if it's a bug but when I follow the usage example in the docs and have turned on `compilerOptions.strict` in `tsconfig.json`, I'm getting a type error: 

```
src/app/shared/store/meta-reducers.ts:17:3 - error TS2322: Type '(state: State, action: Action) => State' is not assignable to type 'ActionReducer<State, Action>'.
  Types of parameters 'state' and 'state' are incompatible.
    Type 'State | undefined' is not assignable to type 'State'.
      Type 'undefined' is not assignable to type 'State'.

17   return metaReducer(reducer);
```
So it seems that `State` can be `undefined` somewhere which is not accepted. If I change `storageSync<State>` to `storageSync<any>` everything seems to work fine. But I guess I'm loosing a bit of type safety then. As a newbie to TypeScript I neither know what the exact problem is nor how to solve it. Maybe something with the type definitons of storageSync()?

## Expected behavior

Should work without type error as described in the docs.

## Minimal reproduction of the problem with instructions

My state:
```
export interface State {
  app: AppState;
}

export const reducers: ActionReducerMap<State> = {
  app: appReducer,
};
```

In my meta-reducer.ts:

```
export function storageSyncReducer(reducer: ActionReducer<State>): ActionReducer<State> {
  const metaReducer = storageSync<State>({
    features: [
      {
        stateKey: 'app', storageForFeature: window.sessionStorage,
        deserialize: (featureState: string) => JSON.parse(featureState)
      },
    ],
    storage: window.localStorage
  });

  return metaReducer(reducer);
}
```

## Environment

<pre>
<code>
@ngrx/store version: 11.0.0
@angular/core: 11.2.0
@typescript: 4.1.5

Browser:
- [x] Chrome (desktop) version 88.0.4324.150
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

**Comment by larscom** [comment]
hi, i will look into this to see if i can reproduce it.

**Comment by larscom** [comment]
Hi, wanted to let you know this is fixed in version 13.0.0+

https://www.npmjs.com/package/@larscom/ngrx-store-storagesync
